src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -1494,6 +1502,66 @@ export class StudyViewPageStore
                                     undefined
                             );
                         }
+                    } else if (this.isGenericAssayChart(chartMeta.uniqueKey)) {
+                        // get generic assay data for the given attribute
+                        // patientAttribute and genericAssayChart are always exist
+                        const isPatientAttribute = chartMeta.patientAttribute;
+                        const genericAssayChart = this._genericAssayChartMap.get(
+                            chartMeta.uniqueKey
+                        )!;
+                        const categorizedGenericAssayProfiles = _.chain(
+                            this.genericAssayProfiles.result
+                        )
+                            .filter(
+                                profile =>
+                                    profile.genericAssayType ===
+                                    genericAssayChart.genericAssayType
+                            )
+                            .groupBy(profile =>
+                                getSuffixOfMolecularProfile(profile)
+                            )
+                            .value();
+                        const sampleMolecularIdentifiers = _.flatMap(
+                            categorizedGenericAssayProfiles[
+                                genericAssayChart.profileType
+                            ].map(profile => profile.molecularProfileId),
+                            molecularId =>
+                                _.map(this.samples.result, sample => {
+                                    return {
+                                        molecularProfileId: molecularId,
+                                        sampleId: sample.sampleId,
+                                    } as SampleMolecularIdentifier;
+                                })
+                        );
+                        data = await defaultClient.fetchGenericAssayDataInMultipleMolecularProfilesUsingPOST(
+                            {
+                                genericAssayDataMultipleStudyFilter: {
+                                    genericAssayStableIds: [
+                                        genericAssayChart.genericAssayEntityId,
+                                    ],
+                                    sampleMolecularIdentifiers,
+                                } as GenericAssayDataMultipleStudyFilter,
+                            } as any
+                        );
+
+                        const keyToData = _.keyBy(data, (d: GenericAssayData) =>
```

Comment:
kinda picky but this would be more performant:    
keyToData = isPatientAttribute ? _.keyBy(...) : _.keyBy(...);